app/ide-desktop/lib/client/src/index.ts

```diff
@@ -38,7 +50,13 @@ class App {
         this.setChromeOptions(chromeOptions)
         security.enableAll()
         electron.app.on('before-quit', () => (this.isQuitting = true))
-        electron.app.whenReady().then(() => this.main(windowSize))
+        // In order to ensure that our `open-url` listener is registered on time, we **must** put it
+        // **before** our `ready listener. We also need to make sure to use `app.on("ready")` and
+        // not `app.whenReady()` because the latter fires earlier than `app.on("open-url")` even if
+        // we register it last.
+        // FIXME [NP]: remove
+        electron.app.on("ready", () => this.main(windowSize))
+        //electron.app.whenReady().then(() => this.main(windowSize))
```

Comment:
1. What is "our open-url listener"? It is not explained anywhere
2. What do you mean by "on time"?
3. There is info that we must put it before something else with no explanation of why.
4. ready listener uses unpaired backticks
5. The last sentence is also not understandable. I'd be thankful if you rewrite the docs to be short, concise , but also explain/justify every statement they make.